### ISSUE TITLE
Bug on list merge

### ISSUE THREAD

**Issue by robsonpeixoto** [akesterson/dpath-python#58]
```python
def test_merge_list():
    src = {"l": [1]}
    p1 = {"l": [2], "v": 1}
    p2 = {"v": 2}
    dst1 = {}
    for d in [src, p1]:
        dpath.util.merge(dst1, d)
    dst2 = {}
    for d in [src, p2]:
        dpath.util.merge(dst2, d)

    assert dst1["l"] == [1, 2]
    assert dst2["l"] == [1]
```

`dst2["l"]` should be `[1]`

`pytest` result:
```
tests/test_util_merge.py::test_merge_list FAILED

========================================================================== FAILURES ==========================================================================
______________________________________________________________________ test_merge_list _______________________________________________________________________

    def test_merge_list():
        src = {"l": [1]}
        p1 = {"l": [2], "v": 1}
        p2 = {"v": 2}
        dst1 = {}
        for d in [src, p1]:
            dpath.util.merge(dst1, d)
        dst2 = {}
        for d in [src, p2]:
            dpath.util.merge(dst2, d)

        assert dst1["l"] == [1, 2]
>       assert dst2["l"] == [1]
E       assert [1, 2] == [1]
E         Left contains more items, first extra item: 2
E         Full diff:
E         - [1, 2]
E         + [1]

tests/test_util_merge.py:123: AssertionError
============================================================= 1 failed, 8 passed in 0.13 seconds =============================================================
```

**Comment by akesterson** [comment]
Thanks for bringing this to our attention. Frustratingly, this is not a bug, it is performing as designed.

But I get that it feels like a bug, and that the behavior is surprising.  See pull #104 to see the clarifying text that I added to the merge() method to help explain this.

We COULD make merge behave differently - but to do so would require us to completely rework the merge method. Currently it begins at the top level of the source/destination objects and traverses downwards, copying as it goes. This would be incompatible with a deepcopy methodology - there might be stuff in the source we don't want, and the only way to avoid that would be to build a list of all paths to be merged, and then begin merging them from the bottom up. That is an expensive modification to an arcane method that we would prefer to avoid. At least in 1.0.

TL;DR - merge() copies references, not objects. Use a deepcopy of the source object to prevent the behavior you are seeing. #104 has an exemplary test.
